tooling/hcpctl/pkg/release/report_test.go

```diff
@@ -0,0 +1,211 @@
+			Namespace:    "aro-hcp",
+			Kind:         "Deployment",
+			DesiredImage: "registry.redhat.io/aro-hcp/backend:v1.2.3",
+			CurrentImage: "registry.redhat.io/aro-hcp/backend:v1.2.3",
+		},
+	}
+	frontendWorkloads := []WorkloadInfo{
+		{
+			Name:         "frontend-deployment",
+			Namespace:    "aro-hcp",
+			Kind:         "Deployment",
+			DesiredImage: "registry.redhat.io/aro-hcp/frontend:v2.1.0",
+			CurrentImage: "registry.redhat.io/aro-hcp/frontend:v2.0.9",
+		},
+	}
+
+	backendRelease := NewComponentRelease("backend", fixedTime, backendWorkloads)
+	backendRelease.Metadata.CreationTimestamp = fixedTime
+
+	frontendRelease := NewComponentRelease("frontend", fixedTime, frontendWorkloads)
+	frontendRelease.Metadata.CreationTimestamp = fixedTime
+
+	reports := []ComponentRelease{backendRelease, frontendRelease}
+
+	tests := []struct {
+		name               string
+		format             string
+		aroHcpCommit       string
+		sdpPipelinesCommit string
+		expectError        bool
+	}{
+		{
+			name:               "YAML output format",
+			format:             "yaml",
+			aroHcpCommit:       "abc123def456",
+			sdpPipelinesCommit: "xyz789uvw012",
+			expectError:        false,
+		},
+		{
+			name:               "JSON output format",
+			format:             "json",
+			aroHcpCommit:       "abc123def456",
+			sdpPipelinesCommit: "xyz789uvw012",
+			expectError:        false,
+		},
+		{
+			name:               "Empty commit values",
+			format:             "yaml",
+			aroHcpCommit:       "",
+			sdpPipelinesCommit: "",
+			expectError:        false,
+		},
+		{
+			name:               "Unsupported format",
+			format:             "xml",
+			aroHcpCommit:       "abc123def456",
+			sdpPipelinesCommit: "xyz789uvw012",
+			expectError:        true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Capture stdout
+			oldStdout := os.Stdout
+			r, w, _ := os.Pipe()
+			os.Stdout = w
+
+			// Mock time.Now() for consistent timestamps in golden files
+			// We'll need to modify the actual output to have a fixed timestamp
+			err := OutputReports(reports, tt.format, tt.aroHcpCommit, tt.sdpPipelinesCommit)
+
+			// Restore stdout and read captured output
+			w.Close()
+			os.Stdout = oldStdout
+			output, _ := io.ReadAll(r)
+
+			if tt.expectError {
+				assert.Error(t, err)
+				return
+			}
+
+			require.NoError(t, err)
+
+			// Parse the output to replace the timestamp with a fixed one for golden comparison
+			var clusterRelease ClusterComponentRelease
+			if tt.format == "json" {
+				require.NoError(t, json.Unmarshal(output, &clusterRelease))
+			} else {
+				require.NoError(t, yaml.Unmarshal(output, &clusterRelease))
+			}
+
+			// Set fixed timestamp for golden comparison
+			clusterRelease.Metadata.CreationTimestamp = fixedTime
+
+			// Use golden fixture for comparison
+			testutil.CompareWithFixture(t, clusterRelease, testutil.WithExtension("."+tt.format))
+		})
+	}
+}
+
+func TestOutputYAML(t *testing.T) {
```

Comment:
Doesn't seem valuable to test the three-line-of-code encoders. One test for output structure is fine